File: Examples/runtimes/java/DynamoDbEncryption/src/test/java/software/amazon/cryptography/examples/keyring/TestMrkDiscoveryMultiKeyringExample.java
package software.amazon.cryptography.examples.keyring;

import java.util.Collections;
import java.util.List;
import java.util.UUID;
import org.testng.annotations.Test;
import software.amazon.cryptography.examples.TestUtils;

public class TestMrkDiscoveryMultiKeyringExample {

  @Test
  public void TestMrkDiscoveryMultiKeyringExample() {
    final String partitionKeyValue = "MrkDiscoveryMultiKeyringExample" + UUID.randomUUID();
    List<String> accounts = Collections.singletonList(
      TestUtils.TEST_AWS_ACCOUNT_ID
    );
    List<String> regions = Collections.singletonList(TestUtils.TEST_AWS_REGION);

    MrkDiscoveryMultiKeyringExample.MultiMrkDiscoveryKeyringGetItemPutItem(
      TestUtils.TEST_DDB_TABLE_NAME,
      TestUtils.TEST_MRK_KEY_ID,
      accounts,
      regions,
      partitionKeyValue,
      "0"
    );
    
    TestUtils.cleanUpDDBItem(
      TestUtils.TEST_DDB_TABLE_NAME,
      "partition_key",
      "sort_key",
      partitionKeyValue,
      "0"
    );
  }
}

File: Examples/runtimes/python/DynamoDBEncryption/test/keyring/test_mrk_discovery_multi_keyring_example.py
# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0
"""Test MRK discovery multi-keyring example."""
import pytest

from ...src.keyring.mrk_discovery_multi_keyring_example import multi_mrk_discovery_keyring_get_item_put_item
from ..test_utils import (
    TEST_AWS_ACCOUNT_ID,
    TEST_AWS_REGION,
    TEST_DDB_TABLE_NAME,
    TEST_MRK_KEY_ID,
)

pytestmark = [pytest.mark.examples]


def test_mrk_discovery_multi_keyring_example():
    """Test mrk_discovery_multi_keyring_example."""
    accounts = [TEST_AWS_ACCOUNT_ID]
    regions = [TEST_AWS_REGION]

    multi_mrk_discovery_keyring_get_item_put_item(TEST_DDB_TABLE_NAME, TEST_MRK_KEY_ID, accounts, regions)
